software/controller/lib/core/alarm.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdint.h>
+#include "vars.h"
+#include "debug.h"
+#include "hal.h"
+#include "alarm.h"
+#include "comms.h"
+#include "commfail_alarm.h"
+
+// MAX_CONTROLLER_ALARMS 
+#define MAX_CONTROLLER_ALARMS 2
+
+// structure definition for MAX_CONTROLLER_ALARMS 
+alarm_cb_t alarms[MAX_CONTROLLER_ALARMS];
+
+// this value is reset to 0 after MAX_CONTROLLER_ALARMS 
+static uint16_t alarm_check_counter;
+
+// There should be a packet from GUI for every 30ms. Assuming something went
+// wrong, we wait for 50ms before raising a communication failure alarm
+static constexpr Duration COMMUNICATION_TIMEOUT = milliseconds(10000);
+static DebugUInt32 dbg_alarm_check("alarm_check",
+                                      "alarm_check",
+                                      0);
+uint32_t alarm_check;
+
+extern Time last_rx;
```

Comment:
This is declared here to be used by `comms.cpp`? This is very confusing.

Reply:
not needed, instead changed the code and using a comms_is_time_to_raise_commfail_alarm() directly form comms.cpp